docroot/modules/custom/annotation_store/src/Controller/AnnotationStoreController.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+namespace Drupal\annotation_store\Controller;
+
+use Drupal\Component\Serialization\Json;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Controller routines for annotation_store routes.
+ */
+class AnnotationStoreController {
+
+  /**
+   * Routing callback - annotation search.
+   */
+  public function annotationStoreSearch() {
+    $this->annotationReqType();
+  }
+
+  /**
+   * Routing callback - annotation save.
+   */
+  public function annotationStoreSave() {
+    $this->annotationReqType();
+  }
+
+  /**
+   * Routing callback - annotation update and delete.
+   */
+  public function annotationStoreUpdateDelete($id) {
+    $this->annotationReqType($id);
+  }
+
+  /**
+   * Get Resource Entity ID.
+   */
+  public function getResourceEntityId() {
+    $request_uri = \Drupal::request()->server->get('HTTP_REFERER');
+    $request_path = parse_url($request_uri, PHP_URL_PATH);
+    // This method is not used in localhost while testing.
+    $path_alias = \Drupal::service('path.alias_storage')->load(array('alias' => $request_path));
+    // If url is from path alias.
+    if (is_array($path_alias)) {
+      $split = explode('/', $path_alias['source']);
+      $resource_entity_id = $split[2];
+    }
+    // If url is without alias.
+    else {
+      $split = explode('/', $request_path);
+      $resource_entity_id = $split[2];
+    }
+    return $resource_entity_id;
+  }
+
+  /**
+   * Annotation search - Returns list of annotations.
+   */
+  public function annotationApiSearch() {
+    $ids = \Drupal::entityQuery('annotation_store')->condition('resource_entity_id', $this->getResourceEntityId())->execute();
+    $obj = entity_load_multiple('annotation_store', $ids);
```

Comment:
entity_load_multiple is deprecated. Use EntityManager to load multiple entities now.